src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/ReassignPropertiesWithDifferentValue.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace DotNetProjectFile.Analyzers.MsBuild;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class ReassignPropertiesWithDifferentValue : MsBuildProjectFileAnalyzer
+{
+    public ReassignPropertiesWithDifferentValue() : base(Rule.ReassignPropertiesWithDifferentValue) { }
+
+    protected override void Register(ProjectFileAnalysisContext context)
+    {
+        if (context.Project.Imports.None()) { return; }
+
+        foreach (var prop in context.Project.PropertyGroups.SelectMany(p => p.Children()))
+        {
+            if (EarlierAssignement(prop, context.Project) is { } previous
+                && Equals(prop.Val, previous.Val))
+            {
+                context.ReportDiagnostic(Descriptor, prop, prop.LocalName);
+            }
+        }
+    }
+
+    private static Node? EarlierAssignement(Node node, MsBuildProject project)
+    {
+        foreach (var import in project.ImportsAndSelf().Reverse().Skip(1))
```

Comment:
This seems quite inefficient. I assume it will not be very problematic with the expected input sizes (famous last words), so if you don't think it's a pressing issue, then perhaps create an issue for this when this gets merged.

Reply:
I defined a reverse too. Because I think it is better that way.